tron/utils/crontab.py

```diff
@@ -37,34 +42,38 @@ class FieldParser:
         re.VERBOSE,
     )
 
-    def normalize(self, source):
+    def normalize(self, source: str) -> str:
         return source.strip()
 
-    def get_groups(self, source):
+    def get_groups(self, source: str) -> List[str]:
         return source.split(",")
 
-    def parse(self, source):
+    def parse(self, source: str) -> Optional[Union[List[int], List[str]]]:
         if source == "*":
             return None
 
-        groups = [self.get_values(group) for group in self.get_groups(source)]
-        groups = set(itertools.chain.from_iterable(groups))
-        has_last = False
-        if "LAST" in groups:
-            has_last = True
+        groups: Set[Union[int, str]] = set(
+            itertools.chain.from_iterable(self.get_values(group) for group in self.get_groups(source))
+        )
+        has_last = "LAST" in groups
+        if has_last:
             groups.remove("LAST")
-        groups = sorted(groups)
+        sorted_groups = sorted(groups, key=lambda x: (isinstance(x, str), x))
         if has_last:
-            groups.append("LAST")
-        return groups
+            sorted_groups.append("LAST")
+        if not sorted_groups:
+            return None
+        if all(isinstance(x, int) for x in sorted_groups):
+            return sorted_groups  # type: ignore
+        return sorted_groups  # type: ignore
```

Comment:
could we add comments for these ignores?

Reply:
Added types. This file was one of those dominoes that came up when adding types elsewhere. I tried to use my whole brain to do this one because it's kind of a spooky change, but I think I got it. Infrastage testing looks good, but it's not exhaustive